exercise/exercise_models.py

```diff
@@ -575,7 +576,15 @@ def _check_submission_allowed(self, profile, request=None):
 
         # Support group id from post or currently selected group.
         group = None
-        group_id = request.POST.get("_aplus_group") if request else None
+        group_id = None
+        if request:
+            try:
+                group_id = json.loads(request.POST.get('__aplus__', '{}')).get('group')
+            except json.JSONDecodeError:
+                pass
```

Comment:
The JSON syntax errors are ignored here in the function `_check_submission_allowed` that is called before any submission is created (in `Submission.objects.create_from_post` that raises an error for JSON syntax errors). This can hide JSON errors in the request. A JSON syntax error would drop the group from the submission and then the submission may be denied due to a required group, so `create_from_post` is never called and the user gets an error about wrong group size or wrong group members, but really there was a syntax error in the JSON of the request.

Reply:
#613